src/components/SearchSelectInput/index.tsx

```diff
@@ -310,7 +310,10 @@ function SearchSelectInput<
             onClearButtonClick={handleClear}
             searchText={searchInputValue}
             onSearchTextChange={handleSearchValueChange}
-            dropdownShown={showDropdown}
+            dropdownShown={showDropdown && (
+                (isTruthyString(searchInputValue) && searchInputValue.length > 0)
+                || realOptions.length > 0
+            )}
```

Comment:
This not not how we hide the dropdown.

We should instead not send a query to the server when we click on the dropdown.